src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ecc.cpp

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_ecc.h"
+
+extern "C" int
+AppleCryptoNative_EccGenerateKey(int32_t keySizeBits, SecKeyRef* pPublicKey, SecKeyRef* pPrivateKey, int32_t* pOSStatus)
+{
+    if (pPublicKey == nullptr || pPrivateKey == nullptr || pOSStatus == nullptr)
+        return kErrorBadInput;
+
+    *pPublicKey = nullptr;
+    *pPrivateKey = nullptr;
+
+    CFMutableDictionaryRef attributes = CFDictionaryCreateMutable(nullptr, 2, &kCFTypeDictionaryKeyCallBacks, nullptr);
+
+    CFNumberRef cfKeySizeValue = CFNumberCreate(nullptr, kCFNumberIntType, &keySizeBits);
+
+    CFDictionaryAddValue(attributes, kSecAttrKeyType, kSecAttrKeyTypeEC);
+    CFDictionaryAddValue(attributes, kSecAttrKeySizeInBits, cfKeySizeValue);
+
+    OSStatus status = SecKeyGeneratePair(attributes, pPublicKey, pPrivateKey);
+
+    CFRelease(attributes);
+    CFRelease(cfKeySizeValue);
+
+    *pOSStatus = status;
+    return status == noErr;
+}
+
+extern "C" uint64_t AppleCryptoNative_EccGetKeySizeInBits(SecKeyRef publicKey)
+{
+    if (publicKey == nullptr)
+    {
+        return 0;
+    }
+
+    size_t blockSize = SecKeyGetBlockSize(publicKey);
+
+    // This seems to be the expected size of an ECDSA signature for this key.
+    // But since Apple uses the DER SEQUENCE(r, s) format the signature size isn't
+    // fixed. It might be trying to encode the biggest the DER value could be:
+    //
+    // 256: r is 32 bytes, but maybe one padding byte, so 33.
+    //      s is 32 bytes, but maybe one padding byte, so 33.
+    //      each of those values gets one tag and one length byte
+    //      35 * 2 is 70 payload bytes for the sequence, so one length byte
+    //      and one tag byte, makes 72.
+    //
+    // 384: r,s are 48 bytes, plus padding, length, and tag: 51
+    //      2 * 51 = 102, requires one length byte and one tag byte, 104.
+    //
+    // 521: neither r nor s can have the high bit set, no padding. 66 content bytes
+    //      plus tag and length is 68.
+    //      2 * 68 is 136, since it's greater than 127 it takes 2 length bytes
+    //      so 136 + 2 + 1 = 139. Looks like they accounted for padding bytes anyways.
+    //
+    // This completely needs to be revisited if Apple adds support for "generic" ECC.
+    //
+    // Word of caution: While seeking meaning in these numbers I ran across a snippet of code
```

Comment:
> Is the API at least consistent \ stable?

Not sure what you mean.  The function name exists for both macOS and iOS, but (supposedly) returns different answers between the two.  Each of the macOS and iOS versions are predictable; just different values.

I think for (256, 384, 521) it's macOS(72, 104, 141) and iOS(32, 48, 66).  iOS is returning "keysize in bytes" (which is lossy as far as generic ECC is concerned), macOS is returning "upper bound of a DER-encoded signature using this key with ECDSA" (which is super lossy)